src/common/functions.rs

```diff
@@ -445,6 +445,64 @@ pub fn decorate_with_currency_tag(
     .into()
 }
 
+/// Format currency by stripping out the decimal if it is 0 and calling
+/// [format_with_commas].
+pub fn strip_decimal_if_zero(num_str: String) -> String {
```

Comment:
This is really tricky. You have to take into account localization (different kinds of number formatting). Fortunately I have already developed a function for this: https://github.com/Granola-Team/mina-block-explorer/blob/73d8ab391a4a1080c99265c3659b9995011e3350/src/common/functions.rs#L848. Could you use this function internally to first normalize the number across all locales and then strip the zero? 

For context, this is how numbers are represented in German locale:

<img width="1920" alt="Screen Shot 2024-10-15 at 8 43 59 AM" src="https://github.com/user-attachments/assets/8bbfd138-3096-470c-8704-5a64e64af587">

Using the normalize function first will make a standard number (see unit tests on how), and then you can go from there.

Reply:
You could also update how this function works: https://github.com/Granola-Team/mina-block-explorer/blob/73d8ab391a4a1080c99265c3659b9995011e3350/src/common/functions.rs#L941